assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -0,0 +1,468 @@
+			streams.push( from( ANALYTICS_4_DIMENSION_OPTIONS[ dimension ] ) );
+		} else {
+			// In case when a dimension is not provided or is not recognized, we use NULL to create a stream (an array) with just one value.
+			streams.push( from( [ null ] ) );
+		}
+	} );
+
+	// This is the list of operations that we apply to the combined stream (array) of dimension values.
+	const ops = [
+		// Convert a dimension value to a row object and generate metric values.
+		map( ( dimensionValue ) => ( {
+			dimensionValues: castArray( dimensionValue ).map( ( value ) => ( {
+				value,
+			} ) ),
+			metricValues: generateMetricValues( validMetrics ),
+		} ) ),
+		// Make sure we take the appropriate number of rows.
+		take( args.limit > 0 ? +args.limit : 90 ),
+		// Accumulate all rows into a single array.
+		reduce( ( rows, row ) => [ ...rows, row ], [] ),
+		// Sort rows if args.orderby is provided.
+		map( ( rows ) =>
+			args.orderby ? sortRows( rows, validMetrics, args.orderby ) : rows
+		),
+	];
+
+	// Process the stream of dimension values and add generated rows to the report data object.
+	zip( ...streams )
+		.pipe( ...ops )
+		.subscribe( ( rows ) => {
+			data.rows = rows;
+			data.rowCount = rows.length;
+
+			// We pretend that the first row contains minimums and the last one maximums because we don't
+			// really need mathematically correct values and can simplify the process of finding this information.
+			data.minimums = [
+				{
+					dimensionValues: dimensions.map( ( dimension ) => {
+						if ( dimension === 'dateRange' ) {
+							return { value: 'date_range_0' };
+						}
+
+						return {
+							value: 'RESERVED_MIN',
+						};
+					} ),
+					metricValues: [ ...( rows[ 0 ]?.metricValues || [] ) ],
+				},
+			];
+			data.maximums = [
+				{
+					dimensionValues: dimensions.map( ( dimension ) => {
+						if ( dimension === 'dateRange' ) {
+							return { value: 'date_range_0' };
+						}
+
+						return {
+							value: 'RESERVED_MAX',
+						};
+					} ),
+					metricValues: [
+						...( rows[ rows.length - 1 ]?.metricValues || [] ),
+					],
+				},
+			];
+
+			// Same here, we pretend that the last row contains totals because we don't need it to be mathematically valid.
+			data.totals = [
+				{
+					dimensionValues: dimensions.map( ( dimension ) => {
+						if ( dimension === 'dateRange' ) {
+							return { value: 'date_range_0' };
+						}
+
+						return {
+							value: 'RESERVED_TOTAL',
+						};
+					} ),
+					metricValues: [
+						...( rows[ rows.length - 1 ]?.metricValues || [] ),
+					],
+				},
+			];
+		} );
+
+	// Set the original seed value for the faker.
+	faker.seed( originalSeedValue );
+
+	return [
+		{
+			dimensionHeaders:
+				args?.dimensions?.map( ( dimension ) => ( {
+					name: dimension,
+				} ) ) || null,
+			metricHeaders: validMetrics.map( ( metric ) => ( {
+				name: metric?.alias || metric?.expression || metric.toString(),
```

Comment:
We should no longer have `alias`, but do have `name` instead. We should however never have a metric with an `expression` but no `name` (as described in https://github.com/google/site-kit-wp/issues/6172#issuecomment-1375638193).

So I think this could be changed as follows:

```suggestion
				name: metric?.name || metric.toString(),
```

Reply:
👍